yt_dlp/extractor/mxplayer.py

```diff
@@ -69,27 +98,52 @@ def _real_extract(self, url):
 
         config_dict = source['config']
         video_dict = source['entities'][video_id]
-        stream_urls = self._get_stream_urls(video_dict)
-
-        title = self._og_search_title(webpage, fatal=True, default=video_dict['title'])
 
-        formats = []
-        headers = {'Referer': url}
-        for stream_name, stream_url in stream_urls:
-            if stream_name == 'dash':
-                format_url = url_or_none(urljoin(config_dict['videoCdnBaseUrl'], stream_url))
-                if not format_url:
-                    continue
-                formats.extend(self._extract_mpd_formats(
-                    format_url, video_id, mpd_id='dash', headers=headers))
-
-        self._sort_formats(formats)
-        info = {
-            'id': video_id,
-            'title': title,
-            'description': video_dict.get('description'),
-            'formats': formats
-        }
+        if video_type == 'movie':
+            title = self._og_search_title(webpage, fatal=True, default=video_dict['title'])
+            stream_urls = self._get_stream_urls(video_dict)
+            formats = []
+            headers = {'Referer': url}
+            for stream_name, stream_url in stream_urls:
+                if stream_name == 'dash':
+                    format_url = url_or_none(urljoin(config_dict['videoCdnBaseUrl'], stream_url))
+                    if format_url:
+                        formats.extend(self._extract_mpd_formats(
+                            format_url, video_id, mpd_id='dash', headers=headers))
+                elif stream_name == 'hls':
+                    format_url = url_or_none(urljoin(config_dict['videoCdnBaseUrl'], stream_url))
+                    if not format_url:
+                        continue
+                    formats.extend(self._extract_m3u8_formats(format_url, video_id, fatal=False))
```

Comment:
```suggestion
                    if not format_url:
                        formats.extend(self._extract_m3u8_formats(format_url, video_id, fatal=False))
```
same as above

Also, are headers not needed here?